engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoMultiType.java

```diff
@@ -94,12 +95,18 @@ public String toString() {
     return "MultiType{" + "types=" + Arrays.toString(types) + '}';
   }
 
+  private static boolean isUnique(Type[] types) {
+    var set = new HashSet<>(Arrays.asList(types));
+    return set.size() == types.length;
+  }
+
   @CompilerDirectives.TruffleBoundary
   final boolean hasIntersectionWith(EnsoMultiType other) {
     var my = new HashSet<>(Arrays.asList(types));
+    int sizeBefore = my.size();
     var their = Arrays.asList(other.types);
     my.removeAll(their);
-    return my.size() < types.length;
+    return my.size() < sizeBefore;
```

Comment:
I don't understand the `their` variable but it appears that this could be rewritten to:
```suggestion
    return my.removeAll(their)
```
and the previous line and `sizeBefore` could be discarded.

Reply:
Ok.

Reply:
Addressed in 232ba475fe76522f1cdaf34f3951dd41fad86495